evennia/utils/hex_colors.py

```diff
@@ -0,0 +1,173 @@
+    _RE_HEX_LONG = '[0-9a-fA-F]{6}'
+    _RE_HEX_SHORT = '[0-9a-fA-F]{3}'
+    _RE_BYTE = '[0-2]?[0-9]?[0-9]'
+    _RE_XTERM_TRUECOLOR = rf'\[([34])8;2;({_RE_BYTE});({_RE_BYTE});({_RE_BYTE})m'
+
+    # Used in hex_sub
+    _RE_HEX_PATTERN = f'({_RE_FG_OR_BG})({_RE_HEX_LONG}|{_RE_HEX_SHORT})'
+
+    # Used for greyscale
+    _GREYS = "abcdefghijklmnopqrstuvwxyz"
+
+    TRUECOLOR_FG = f'\x1b\[38;2;{_RE_BYTE};{_RE_BYTE};{_RE_BYTE}m'
+    TRUECOLOR_BG = f'\x1b\[48;2;{_RE_BYTE};{_RE_BYTE};{_RE_BYTE}m'
+
+    # Our matchers for use with ANSIParser and ANSIString
+    hex_sub = re.compile(rf'{_RE_HEX_PATTERN}', re.DOTALL)
+
+    def sub_truecolor(self, match: re.Match, truecolor=False) -> str:
+        """
+        Converts a hex string to xterm truecolor code, greyscale, or
+        falls back to evennia xterm256 to be handled by sub_xterm256
+
+        Args:
+            match (re.match): first group is the leading indicator,
+                              second is the tag
+            truecolor (bool): return xterm truecolor or fallback
+
+        Returns:
+            Newly formatted indicator and tag (str)
+
+        """
+        indicator, tag = match.groups()
+
+        # Remove the # sign
+        indicator = indicator.replace('#', '')
+
+        r, g, b = self._hex_to_rgb_24_bit(tag)
+
+        # Is it greyscale?
+        if r == g and g == b:
+            return f"{indicator}=" + self._GREYS[self._grey_int(r)]
+
+        else:
+            if not truecolor:
+                # Fallback to xterm256 syntax
+                r, g, b = self._rgb_24_bit_to_256(r, g, b)
+                return f"{indicator}{r}{g}{b}"
+
+            else:
+                xtag = f"\033["
+                if '[' in indicator:
+                    # Background Color
+                    xtag += '4'
+
+                else:
+                    xtag += '3'
+
+                xtag += f"8;2;{r};{g};{b}m"
+                return xtag
+
+    def xterm_truecolor_to_html_style(self, fg="", bg="") -> str:
+        """
+        Converts xterm truecolor to an html style property
+
+        Args:
+            fg: xterm truecolor
+            bg: xterm truecolor
+
+        Returns: style='color and or background-color'
+
+        """
+        prop = 'style="'
+        if fg != '':
+            res = re.search(self._RE_XTERM_TRUECOLOR, fg, re.DOTALL)
+            fg_bg, r, g, b = res.groups()
+            r = hex(int(r))[2:].zfill(2)
+            g = hex(int(g))[2:].zfill(2)
+            b = hex(int(b))[2:].zfill(2)
+            prop += f"color: #{r}{g}{b};"
+        if bg != '':
+            res = re.search(self._RE_XTERM_TRUECOLOR, bg, re.DOTALL)
+            fg_bg, r, g, b = res.groups()
+            r = hex(int(r))[2:].zfill(2)
+            g = hex(int(g))[2:].zfill(2)
+            b = hex(int(b))[2:].zfill(2)
+            prop += f"background-color: #{r}{g}{b};"
+        prop += f'"'
+        return prop
+
+    def _split_hex_to_bytes(self, tag: str) -> tuple[str, str, str]:
```

Comment:
Minor style-nitpick: Place the methods being used _before_ the methods using them. This is standard across Evennia. So move these private methods up towards the top of this class.